test/bedrock_Azure_apim_waf_mc_test.go

```diff
@@ -0,0 +1,266 @@
+	}
+	eastIP, err2 := ioutil.ReadFile("../cluster/environments/azure-multiple-clusters-waf-tm-apimgmt/output/" + eastIP_address_file)
+	if err2 != nil {
+		panic(err2)
+	}
+	centralIP, err3 := ioutil.ReadFile("../cluster/environments/azure-multiple-clusters-waf-tm-apimgmt/output/" + centralIP_address_file)
+	if err3 != nil {
+		panic(err3)
+	}
+
+	fmt.Println("West Cluster IP: " + string(westIP))
+	fmt.Println("East Cluster IP: " + string(eastIP))
+	fmt.Println("Central Cluster IP: " + string(centralIP))
+
+	//Deploy app to all 3 clusters
+	configFile := "azure-vote.yaml"
+
+	addIPtoYAMLAPIM(configFile, string(westIP))
+	k8s.KubectlApply(t, options, configFile)
+	addIPtoYAMLAPIM(configFile, string(eastIP))
+	k8s.KubectlApply(t, options2, configFile)
+	addIPtoYAMLAPIM(configFile, string(centralIP))
+	k8s.KubectlApply(t, options3, configFile)
+
+	// Test Case 4: Validate API Management - API created for Voting
+	if len(clientid) > 0 {
+		log.Printf("Setting client ID")
+		os.Setenv("AZURE_CLIENT_ID", clientid)
+	}
+	if len(clientsecret) > 0 {
+		log.Printf("Setting client secret")
+		os.Setenv("AZURE_CLIENT_SECRET", clientsecret)
+	}
+	if len(tenantid) > 0 {
+		log.Printf("Setting Tenant ID")
+		os.Setenv("AZURE_TENANT_ID", tenantid)
+	}
+	if len(subscriptionid) > 0 {
+		log.Printf("Setting Subscription ID")
+		os.Setenv("AZURE_SUBSCRIPTION_ID", subscriptionid)
+	}
+	apim := apimanagement.NewAPIClient(subscriptionid)
+	authorizer, err := auth.NewAuthorizerFromEnvironment()
+	if err == nil {
+		apim.Authorizer = authorizer
+	} else {
+		t.Log("Couldnt find environment variables for client credentials, initiatig Azure CLI ...")
+		authorizer, err1 := auth.NewAuthorizerFromCLI()
+		if err1 == nil {
+			apim.Authorizer = authorizer
+		}
+	}
+	apicontract, err := apim.Get(ctx, resourcegroupname, apiservicename, apiname)
+
+	if err != nil {
+		fmt.Println("Error validating API Management - Removing cluster load balancer and Destroying resources")
+		_clean, cleanErr := k8s.RunKubectlAndGetOutputE(t, options, "delete", "service", "azure-vote-front")
+		if cleanErr != nil || !strings.Contains(_clean, "delete") {
+			t.Fatal(cleanErr)
+		} else {
+			fmt.Println("Clean verification for West Cluster complete")
+		}
+		_clean2, cleanErr2 := k8s.RunKubectlAndGetOutputE(t, options2, "delete", "service", "azure-vote-front")
+		if cleanErr2 != nil || !strings.Contains(_clean2, "delete") {
+			t.Fatal(cleanErr2)
+		} else {
+			fmt.Println("Clean verification for East Cluster complete")
+		}
+		_clean3, cleanErr3 := k8s.RunKubectlAndGetOutputE(t, options3, "delete", "service", "azure-vote-front")
+		if cleanErr3 != nil || !strings.Contains(_clean3, "delete") {
+			t.Fatal(cleanErr3)
+		} else {
+			fmt.Println("Clean verification for Central Cluster complete")
+		}
+		//Sleep job for 2 minutes while load balancer deallocates
+		time.Sleep(120 * time.Second)
+		t.Fatal(cleanErr)
+	} else {
+
+		name := *apicontract.Name
+		fmt.Println("Test Passed API created  ")
```

Comment:
With this test can we also assume WAF has been deployed successfully? @shwetams

Reply:
@NathanielRose  - this is what we discussed since there is a manual step involved between linking APIM with WAF. The only way to test WAF is to have the test running inside the VNet. We had discussed that this is something we will document and work later on.